tools/tpm2_tool.c

```diff
@@ -134,6 +135,8 @@ int main(int argc, char *argv[], char *envp[]) {
     /* TODO SAPI INIT */
     TSS2_SYS_CONTEXT *sapi_context = sapi_ctx_init(tcti);
 
+    tpm2_errata_init(sapi_context);
```

Comment:
A failure here should minimally print a warning message that errata application cannot be applied. The fix routine on this un-initialized state should warn what errata fixup was not applied and return success. Ie try and continue in the face of errors until the tpm gives us a hard stop.

Reply:
If tpm2_errata_init fails, this should be a critical issue, because one of items in TPM_PT_LEVEL, TPM_PT_REVISION, TPM_PT_DAY_OF_YEAR or TPM_PT_YEAR is incorrect in hardware level. But maybe such a tpm can still work as expected. So I would prefer to prompt a warning for this failure.

For tpm2_errata_name and tpm2_errata_fixup, returning false means the requesting errata cannot be applied due to some reasons, one of which includes the same reason where tpm2_errata_init fails.

Therefore, I prefer tpm2_errata_fixup without return value, and let tpm2_errata_fixup print the "Errata %s applied".